src/org/labkey/targetedms/outliers/CUSUMOutliers.java

```diff
@@ -669,10 +669,11 @@ public JSONObject getSampleFilesJSON(Map<String, SampleFileInfo> sampleFiles)
 
         for (LJOutlier ljOutlier : ljOutliers)
         {
-            if(sampleFileInfo == null || (!(ljOutlier.getSampleFile() != null && ljOutlier.getSampleFile().equals(sampleFileInfo.getSampleFile()))))
+            String sampleFileString = ljOutlier.getSampleFile() + "_" + ljOutlier.getAcquiredTime();
```

Comment:
Why are you concatenating these together? To make sure it's unique? Would it be better to use the Id value from targetedms.samplefile?